soc/arm/infineon_cat1/psoc6/Kconfig.soc

```diff
@@ -0,0 +1,117 @@
+# Copyright (c) 2021 Cypress Semiconductor Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+# Infineon PSoC6 die
+config SOC_DIE_PSOC6
+	bool
+	select ARM
+	select CPU_CORTEX_M4
+	select CPU_HAS_ARM_MPU
+	select HAS_SRAM_OFFSET
+	select SOC_FAMILY_CAT1A
+	select DYNAMIC_INTERRUPTS
+
+# Infineon PSoC6_01 die
+config SOC_DIE_PSOC6_01
+	bool
+	select SOC_DIE_PSOC6
+
+# Infineon PSoC6_02 die
+config SOC_DIE_PSOC6_02
+	bool
+	select SOC_DIE_PSOC6
+
+# Infineon PSoC6_03 die
+config SOC_DIE_PSOC6_03
+	bool
+	select SOC_DIE_PSOC6
+
+# Infineon PSoC6_04 die
+config SOC_DIE_PSOC6_04
+	bool
+	select SOC_DIE_PSOC6
+
+# Infineon soc packages
+config SOC_PACKAGE_PSOC6_01_124_BGA
+	bool
+
+config SOC_PACKAGE_PSOC6_01_116_BGA_BLE
+	bool
+
+config SOC_PACKAGE_PSOC6_01_104_M_CSP_BLE
+	bool
+
+config SOC_PACKAGE_PSOC6_01_80_WLCSP
+	bool
+
+config SOC_PACKAGE_PSOC6_01_116_BGA_USB
+	bool
+
+config SOC_PACKAGE_PSOC6_01_124_BGA_SIP
+	bool
+
+config SOC_PACKAGE_PSOC6_01_43_SMT
+	bool
+
+config SOC_PACKAGE_PSOC6_01_104_M_CSP_BLE_USB
+	bool
+
+config SOC_PACKAGE_PSOC6_01_68_QFN_BLE
+	bool
+
+config SOC_PACKAGE_PSOC6_02_124_BGA
+	bool
+
+config SOC_PACKAGE_PSOC6_02_128_TQFP
+	bool
+
+config SOC_PACKAGE_PSOC6_02_100_WLCSP
+	bool
+
+config SOC_PACKAGE_PSOC6_02_68_QFN
+	bool
+
+config SOC_PACKAGE_PSOC6_03_100_TQFP
+	bool
+
+config SOC_PACKAGE_PSOC6_03_68_QFN
+	bool
+
+config SOC_PACKAGE_PSOC6_03_49_WLCSP
+	bool
+
+config SOC_PACKAGE_PSOC6_04_64_TQFP
+	bool
+
+config SOC_PACKAGE_PSOC6_04_68_QFN
+	bool
+
+config SOC_PACKAGE_PSOC6_04_80_TQFP
+	bool
+
+## Infineon MCUs
+choice
+	prompt "MPN"
+	osource "soc/arm/infineon_cat1/psoc6/Kconfig.soc.psoc6_*"
+endchoice
+
+if SOC_FAMILY_CAT1A
+## PSoC™ 6 Cortex M0+ prebuilt images
+choice
+	prompt "PSoC™ 6 Cortex M0+ prebuilt images"
+	help
+	  Prebuilt application images are executed on the Cortex-M0+ core of the PSoC™ 6 dual - core MCU.
+	  The images are provided as C arrays ready to be compiled as part of the Cortex-M4 application.
+	  The Cortex-M0+ application code is placed to internal flash by the Cortex-M4 linker script.
+
+config SOC_PSOC6_CM0P_SLEEP
+	bool "DeepSleep"
```

Comment:
This is definitely is not the correct approach/name. I mean, those MCUs are asymmetric dual core. There is no problem to build Cortex-M4 with buildin CM0+ but this name is not good and don't tell to anyone the real meaning.

In the current implementation M0+ is always available and M4 can be enabled by user. https://github.com/zephyrproject-rtos/zephyr/blob/b9e833366f1360ecbad8776d6e8587e180688e1b/soc/arm/cypress/Kconfig#L24-L28
This is an small convenience because M0+ boots M4.

I think a similar strategy could be made here where user can decide if he wants a prebuild image integrated or he will build a separeted image to both Cortex-M0+ and M4. This could be a hidden symbol initially to always support the prebuild image (default).

Reply:
@nandojve, 
per https://github.com/zephyrproject-rtos/zephyr/issues/42883 section "Cortex-M0+ support":
_"In the PSoC 6-based integration, the family's Cortex-M0+ will be controlled by a pre-built image, rather than being open to the user. This aligns with our current PSoC 6 strategy. This pre-built image will be acquired from the Infineon MCU SDK via west extensions the user will invoke when creating a PSoC 6-based application. In future revisions of our Zephyr support, including when adding new device families, we will seek to open the Cortex-M0+ for user development."_

i ok to rename: 
SOC_PSOC6_CM0P_SLEEP -> SOC_PSOC6_CM0P_IMAGE_SLEEP
SOC_PSOC6_CM0P_****** -> SOC_PSOC6_CM0P_IMAGE_****

@ifyall, please provide your thoughts.

Reply:
A rename to make the intent clear makes sense to me.
Regarding the prompt, perhaps we can change it to something like this:

`choice
	prompt "PSoC™ 6 Cortex M0+ prebuilt images"
	help
	  Choose the prebuilt application image to be executed on the Cortex-M0+ core of the PSoC™ 6 dual - core MCU. The image is responsible for booting the Cortex-M4 on the device.`

Reply:
done